Adding-new-translations-for-i18n.md

```diff
@@ -33,12 +52,138 @@ In the example above, if there is only one notification, then we should change "
+
+When updating the English text for a translation key, first determine whether the new English text has a significantly different meaning to the previous English text.
+
+- If the new text conveys a different meaning from the previous text: do not reuse the translation key. Create a new translation key instead, and delete the previous one.
+- If the new text is similar to the previous text, and previous translations are probably still valid: just update the English value directly in `assets/i18n/en.json`. Translatewiki will pick up the updates and push new translations in due course.
+
+### Deleting removed translation keys
+
+If a translation key is not used any more, you must delete it from `assets/i18n/en.json` **and all other translation JSON files**. This is to preserve the property that the keys in other JSON files are a subset of those in `en.json`.
+
+### Verifying changes to translation files
+
+To verify your changes to translation files locally, run the following command in a terminal:
+
+Python:
+```
+python -m scripts.run_backend_tests --test_targets=core.controllers.base_test.I18nDictsTests
+```
+
+Docker:
+```
+make run_tests.backend PYTHON_ARGS="--test_targets=core.controllers.base_test.I18nDictsTests"
+```
+
+This validates the translation JSON files by verifying that the keys are correctly sorted, that the keys in en.json and qqq.json match, that every other translation JSON file has a subset of the keys in en.json, and so on.
+
+
+### Flash of Untranslated Content
+
+Sometimes, the page is rendered in the browser before the locale file with the translations is loaded. As a result, the user can see the translation keys briefly before they're replaced with the corresponding translations, which is not a good user experience. This problem is known as the "Flash of Untranslated Content", or FoUC.
+
+When adding a new string to Oppia's HTML code, please take into account the following tips to prevent FoUC:
+- The FoUC behaves differently in the preferred language (English) and all the other languages. Please manually check that there is no FoUC in both cases.
+- Add the translation key inside the html tag as the value of the translate attribute. This will prevent the key from being shown briefly -- instead, the location of the string will remain empty in the interim.
+- If the string is in a very visible location and there is FoUC in the preferred language, add the key and the translation into the `DEFAULT_TRANSLATIONS` constant defined in the file [i18n.js](https://github.com/oppia/oppia/blob/develop/core/templates/i18n.js).
+
+### Titles
+
+To add a title to the window, replace the block maintitle for a translation key:
+
+    {% block maintitle %}TRANSLATE_KEY{% endblock maintitle %}
+
+and add the translation into the corresponding json files. Titles should always be translated, but this is not mandatory. If you replace this block with a common string, it will be shown as the title instead.
```

Comment:
why do we allow untranslated titles?

Reply:
I actually just copied this from the previous doc, but ... wow, this is really old, this is the Jinja syntax from, like, 2016 :P I deleted it entirely since it no longer applies!